### PR TITLE
Introduce IndexSet::get_index_vector().

### PR DESCRIPTION
**Pull request by bangerth**
This is in response to some code I saw in #14736. This is the more "deal.II way" these days.

@stefanozampini FYI

### CODE REVIEW

File: include/deal.II/base/index_set.h [hunk 1]
```diff
@@ -385,8 +385,22 @@ class IndexSet
...
+  get_index_vector() const;
+
   /**
    * Fill the given vector with all indices contained in this IndexSet.
+   *
+   * This function is equivalent to calling get_index_vector() and
+   * assigning the result to the @p indices argument.
    */
   void
   fill_index_vector(std::vector<size_type> &indices) const;
```
Comment (drwells) [inline include/deal.II/base/index_set.h:406]: We deprecated the other fill functions - should this be deprecated too?

Comment (bangerth) [reply include/deal.II/base/index_set.h:406]: I had no specific opinion about this, but am happy to do that if you'd like. Given that there are two thumbs up on your comment, I take that as a yes.

Comment (drwells) [reply include/deal.II/base/index_set.h:406]: Please do!

**Comment by bangerth** [comment]
I deprecated the function and replaced it in all tests where it is used.

**Review by drwells — APPROVED** [review]

**Pull request merged**